specs/trip_dispatch_spec.rb

```diff
@@ -88,5 +89,61 @@
       passenger.must_be_instance_of RideShare::Passenger
       passenger.trips.must_include trip
     end
+
+  end
+
+  describe "request_trip method" do
+    before do
+      @dispatcher = RideShare::TripDispatcher.new
+    end
+
+    it "creates a new instance of trip" do
+      requested_trip = @dispatcher.request_trip(1)
+
+      requested_trip.must_be_instance_of RideShare::Trip
+    end
+
+    it "updates trip lists for driver, passenger, and tripdispatcher" do
+      # this assumes that the driver selected is driver 2
+      dispatcher_trips = @dispatcher.trips.length
+      driver = @dispatcher.find_driver(2)
+      driver_trips = driver.trips.length
+      passenger = @dispatcher.find_passenger(1)
+      passenger_trips = passenger.trips.length
+
+      new_trip = @dispatcher.request_trip(1)
+
+      @dispatcher.trips.length.must_equal dispatcher_trips + 1
+      new_trip.driver.trips.length.must_equal driver_trips + 1
+      new_trip.passenger.trips.length.must_equal passenger_trips + 1
+    end
+
+    it "supplies a passenger id as parameter" do
+      requested_trip = @dispatcher.request_trip(1)
+
+      requested_trip.passenger.must_be_instance_of RideShare::Passenger
+    end
+
+    it "automatically assigns first available driver" do
+      requested_trip = @dispatcher.request_trip(1)
+
+      requested_trip.driver.must_be_instance_of RideShare::Driver
+      requested_trip.driver.status.must_equal :AVAILABLE
```

Comment:
Why would the driver now be `:AVAILABLE`?  Since the driver has just gotten a trip, it should be `:UNAVAILABLE`